sycl/source/detail/pi.cpp

```diff
@@ -61,10 +101,22 @@ void initialize() {
   if (!useBackend(SYCL_BE_PI_OPENCL)) {
     die("Unknown SYCL_BE");
   }
-  #define _PI_API(api)                          \
-    extern decltype(::api) * api##OclPtr; \
-    api = api##OclPtr;
-  #include <CL/sycl/detail/pi.def>
+
+  std::string PluginPath = findPlugin();
+  if (PluginPath.empty())
+    die("Plugin Not Found.");
```

Comment:
Why are we "die" here? Shouldn't we just return saying "no opencl plugins/platforms" available to the caller?

Reply:
By return do you mean return and halt the program execution?
The function die currently, prints a message and calls std::terminate.

Reply:
No I mean just return an error. SYCL RT can continue working with other plugins.